p2p/switch.go

```diff
@@ -401,28 +416,41 @@ func isPrivateAddr(err error) bool {
 	return ok && te.PrivateAddr()
 }
 
-// DialPeersAsync dials a list of peers asynchronously in random order (optionally, making them persistent).
+// DialPeersAsync dials a list of peers asynchronously in random order.
 // Used to dial peers from config on startup or from unsafe-RPC (trusted sources).
-// TODO: remove addrBook arg since it's now set on the switch
-func (sw *Switch) DialPeersAsync(addrBook AddrBook, peers []string, persistent bool) error {
+// It ignores ErrNetAddressLookup. However, if there are other errors, first
```

Comment:
Imagine you're starting a k8s cluster and DNS does not work yet => lookup fails => Tendermint fails to start.

If we ignore ErrNetAddressLookup, we won't be able to dial peers immediately, but reconnect will connect us to them eventually (only to persistent peers though).

Maybe we should add a flag `ignoreNetAddressLookupErrors`

Reply:
> but reconnect will connect us to them eventually (only to persistent peers though).

this won't work in the current implementation because addr is `nil`. hmmm

Reply:
https://github.com/tendermint/tendermint/issues/880

Reply:
I think NetAddress should resolve IP right before dialing + resolve it again when IP is no longer reachable.